routers/web/repo/editor.go

```diff
@@ -64,13 +69,13 @@ func redirectForCommitChoice(ctx *context.Context, commitChoice, newBranchName,
 		repo := ctx.Repo.Repository
 		baseBranch := ctx.Repo.BranchName
 		headBranch := newBranchName
-		if repo.UnitEnabled(ctx, unit.TypePullRequests) {
-			redirectToPullRequest = true
-		} else if canCreateBasePullRequest(ctx) {
+		if canCreateBasePullRequest(ctx) {
 			redirectToPullRequest = true
 			baseBranch = repo.BaseRepo.DefaultBranch
 			headBranch = repo.Owner.Name + "/" + repo.Name + ":" + headBranch
 			repo = repo.BaseRepo
+		} else if repo.UnitEnabled(ctx, unit.TypePullRequests) {
+			redirectToPullRequest = true
```

Comment:
I don't quite understand the change, and it doesn't match the comment "// Redirect to a pull request when possible" above now.

Please add more comments for the details and it's better to have some tests to clarify the behavior.

Reply:
I added some comments now.

Note that whether or not it redirects to the pull request is not changed by this. Just which repository and branch is the default choice in the pull request.

For testing I'll have a look later.